### PR TITLE
adding new EP for get user

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
i tried :)

Dual Merge: [FE](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/1060) & [BE](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/409)

migrations and seeds so, if inside mock dir:
```
docker-compose run app npm run migrate
docker-compose run app npm run seed 
```

### CODE REVIEW

File: src/routes.js [hunk 1]
```diff
@@ -170,6 +170,10 @@ var appRouter = function (app) {
   });
 
   app.get('/availablePositions', async function(req, res) {
+  print('--------------------aahsbgkjahkbs kgjhjbh gkjhjbdhjbdsdfksjbdfkjbg')
```
Comment (elizabeth-jimenez) [inline src/routes.js:173]: 🤪  someone was losing their mind

**Comment by elizabeth-jimenez** [comment]
@mjoyce91 @scott062 i think this is ok, except it is working on the assumption that hru_id === perdet_seq_num. I'm going to dbl check that in devnet, but i think the rest is good to go 🤞 🤞

**Comment by scott062** [comment]
> @mjoyce91 @scott062 i think this is ok, except it is working on the assumption that hru_id === perdet_seq_num. I'm going to dbl check that in devnet, but i think the rest is good to go 🤞 🤞
> 
> update: well, within our local, the user can have both hru_id and perdet_seq_num, and they aren't the same.

To clarify: hru_id is not the perdet, but they are unique employee identifiers. Ideally we could use either to filter but we want to check devnet. I believe the payload has perdet coming back as emp_seq_num which is odd. So filtering on perdet will show up as emp_seq_num in payload last time i checked

File: src/services/employees.js [hunk 2]
```diff
@@ -200,6 +200,13 @@ const get_persons_filters = (params = {}) => {
   return q
 }
 
+const get_user_filters = (params = {}) => {
+  const { hru_id } = params
+  const q = {}
+  if (hru_id) q['employees.hru_id'] = hru_id
+  return q
+}
+
```
Comment (scott062) [inline src/services/employees.js:209]: add perdet? and does it expect `request_params.` or not?

File: src/services/employees.js [hunk 3]
```diff
@@ -453,4 +471,79 @@ const get_persons = async query => {
...
+            gal_update_id: user.gal_update_id || '',
+            gal_update_date: user.gal_update_date || '',
+            gal_fax_nbr_text: user.gal_fax_nbr_text || '',
+            gal_mobile_nbr_text: user.gal_mobile_nbr_text || '',
+            gal_pager_nbr_text: user.gal_pager_nbr_text || '',
+            gal_home_nbr_text: user.gal_home_nbr_text || '',
+            gal_notes_text: user.gal_notes_text || '',
+            gal_domain_name: user.gal_domain_name || '',
+            rnum: user.rnum || ''
+        }
```
Comment (scott062) [inline src/services/employees.js:539]: @mjoyce91 Now that they're giving us a ton more info, we may want to see if any of the override fields they're sending are relevant for other information we're displaying

File: src/services/employees.js (hunk 2 above)
Comment (mjoyce91) [reply src/services/employees.js:209]: yeah, what's this doing? Is this just to map `hru_id` to `employees.hru_id`? And same question as Scott for `request_params.`

Comment (elizabeth-jimenez) [reply src/services/employees.js:209]: Pretty sure this is setting up to filter on `hru_id`

**Comment by scott062** [comment]
@mjoyce91 Are we waiting on this till after oracle merge?

**Comment by elizabeth-jimenez** [comment]
> @mjoyce91 Are we waiting on this till after oracle merge?

😭

**Comment by mjoyce91** [comment]
No this can be merged independent of Oracle change

File: src/services/employees.js [hunk 4]
```diff
@@ -202,31 +202,42 @@ const get_persons_filters = (params = {}) => {
...
+            } else {
+                // Default sort
+                qb.orderBy('employees.last_name')
+            }
+        }
+    })
+}
+
+// Query for fetching users
+const get_users_query = (params, mapping) => {
```
Comment (scott062) [inline src/services/employees.js:234]: Don't need mapping

File: src/services/employees.js [hunk 5]
```diff
@@ -202,31 +202,42 @@ const get_persons_filters = (params = {}) => {
...
+        const isCount = params['request_params.get_count'] === 'true'
+        if (!isCount) {
+            const orderByField = params['request_params.order_by']
+            if (orderByField) {
+                addOrderBy(qb, orderByField, SORT_MAPPING)
+            } else {
+                // Default sort
+                qb.orderBy('employees.last_name')
+            }
+        }
```
Comment (scott062) [inline src/services/employees.js]: Indent

**Review by scott062 — APPROVED** [review]
☎️  🏠  🔪

**Pull request merged**